uwblueprint/starter-code-cli
# Add initial test script

## Notion ticket link



[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)



## Implementation description

- Currently tests are sequential - we discussed pre-loading data earlier, but I opted not to do that for the first iteration since we need to pre-load firebase data as well, need to figure out a way to pre-load data that doesn’t just mimic the starter code implementation 
- It does tell you where the test fails, but it will just skip the rest of the tests for that suite
- So these are more like end-to-end tests for each group of endpoints
- One benefit is that they are self-cleaning (it will create data in the database and delete all of it when the test finishes)



## Steps to test

1. Add EMAIL=<login_email> and PASSWORD=<password> to root .env file (your starter code login credentials)
2. Run `docker-compose up --build` in the starter-code-v2 repo
3. Run pytest in the tests folder of this repo, with command line arguments according to the stack combination
--lang python (if running Python)
--api graphql (if running GraphQL)
--auth true (if using auth)
--fs true (if using file storage)
4. Verify that all test suites pass, and that no new data is left in the database




## What should reviewers focus on?

- Is there any part that can be written in a cleaner way?
- Not sure how to do logout for a new user since they need to verify their email first, should we instead logout the admin user we run the script with? (requires either enforcing an order so logout comes last or performing login before each test suite)

## Checklist
